### DISCUSSION TITLE
Error: assignment to contiguous pointer from non-contiguous target at (1)

### DISCUSSION THREAD

**Discussion by Julia-Cheng0225** [fabm-model/fabm#35] [Q&A] [answered]
[ 86%] Built target fabm_models_au
[ 88%] Built target fabm_models_csiro
[ 89%] Building Fortran object CMakeFiles/fabm.dir/src/fabm.F90.obj
C:\fabm\src\fabm.F90:1360:17:

          pdat => dat(_PREARG_LOCATION_DIMENSIONS_ i)
                 1
Error: Assignment to contiguous pointer from non-contiguous target at (1)
mingw32-make[2]: *** [CMakeFiles\fabm.dir\build.make:173: CMakeFiles/fabm.dir/src/fabm.F90.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:394: CMakeFiles/fabm.dir/all] Error 2
mingw32-make: *** [Makefile:135: all] Error 2

**Comment by Julia-Cheng0225** [comment]
Hi, I encountered an issue while trying to follow the steps outlined in "MinGW-w64: GNU Fortran on Windows" and am unable to resolve it.I would be grateful if someone could offer assistance and help me resolve this issue！！

**Comment by jornbr** [comment, marked as answer]
I suspect this happens because you are using version 8 of MinGW (gfortran). This version is too strict in its checking of array contiguity. You can work around this issue by switching `FABM_USE_CONTIGUOUS` to `OFF` in cmake (see also https://github.com/fabm-model/fabm/wiki/Building-and-installing#supported-compilers). And if you are indeed using version 8.x of MinGW, you could also upgrade to a later version instead.

**Reply by Julia-Cheng0225** [reply]
Thank you for helping! I tried these methods but still going through the same problem. I think I'm gonna try to use Linux to install fabm and hope it can work well! !

**Reply by jornbr** [reply]
Ok. Just to note: if `FABM_USE_CONTIGUOUS` is truly `OFF`, there are no contiguous arrays used by FABM at all, which means that this specific error ("Assignment to contiguous pointer...") can no longer occur. Thus, I suspect that `FABM_USE_CONTIGUOUS=OFF` was not recognized by cmake. Note that if you are using CMake-GUI, you'll have to first check "Advanced", and then you can switch off  `FABM_USE_CONTIGUOUS`:

![image](https://user-images.githubusercontent.com/2621530/232424721-9cbf2db0-e531-46a5-b171-cbb25d4f5c13.png)

After that, you'll have to click "Configure" and "Generate" again before you rebuild.

**Reply by Julia-Cheng0225** [reply]
Thank you for your patient help! This time it works. I only changed FABM_USE_CONTIGUOUS in the CMakelists file, and after checking in CMake-GUI the installation worked! Thank you so much!!
